### PR TITLE
mysql8: new port for MySQL 8.0.16

### PR DESCRIPTION
**Pull request by herbygillot**
#### Description

MySQL 8 has been out since April 2018, but for some reason never made it into MacPorts.  This is my stab at putting a port together for it.  This could benefit from some extra eyes and additional testing.

Notes:
  * `mysqld` refused to have a correct built-in basedir no matter what I did during compilation.  So I am currently setting `basedir` in configuration (`macports-default.cnf`)
  * I have not yet tried to compile anything against MySQL 8 libraries and headers, so that could stand for some testing.
  * MySQL 8.0.16 introduces a new piece of software, [MySQL Router](https://dev.mysql.com/doc/mysql-router/8.0/en/)
    * We are currently *not* building Router, because when we do, MacPorts marks the build as failed due to library wonkiness detected in the `mysqlrouter` binary
    * MySQL Router should probably have its own subport
    * MySQL Router does not seem to respect a lot of the CMake path configuration vars
  * It's possible I've overlooked a binary, man page or other resource that may need to be added to the `mysql-select` manifest
  
Other than the above, installing and following the notes for `mysql8` and `mysql8-server` seems to work as expected.

Related Trac tickets:
    * https://trac.macports.org/ticket/57776
    * https://trac.macports.org/ticket/56420

###### Tested on

macOS 10.14.5 18F203
Xcode 10.2.1 10E1001

###### Verification 
Have you

- [x] followed our [Commit Message Guidelines](https://trac.macports.org/wiki/CommitMessages)?
- [x] squashed and [minimized your commits](https://guide.macports.org/#project.github)?
- [x] checked that there aren't other open [pull requests](https://github.com/macports/macports-ports/pulls) for the same change?
- [x] referenced existing tickets on [Trac](https://trac.macports.org/wiki/Tickets) with full URL?

- [x] checked your Portfile with `port lint`?
- [x] tried a full install with `sudo port -vst install`?
- [x] tested basic functionality of all binary files?

### CODE REVIEW

**Comment by herbygillot** [comment]
Not sure why the builds are failing at the `fetch` step, seems to work here:

```
hephaestus:macports-ports herby$ sudo port fetch mysql8
--->  Fetching distfiles for mysql8
--->  Attempting to fetch mysql-8.0.16.tar.gz from https://distfiles.macports.org/mysql8
--->  Attempting to fetch mysql-8.0.16.tar.gz from http://mirror.trouble-free.net/mysql_mirror/Downloads/MySQL-8.0
--->  Attempting to fetch mysql-8.0.16.tar.gz from http://mysql.mirrors.ovh.net/ftp.mysql.com/Downloads/MySQL-8.0
--->  Attempting to fetch boost_1_69_0.tar.gz from https://distfiles.macports.org/mysql8
--->  Attempting to fetch boost_1_69_0.tar.gz from https:///dl.bintray.com/boostorg/release/1.69.0/source
hephaestus:macports-ports herby$ echo $?
0
```

**Comment by cjones051073** [comment]
Thanks for this.

Looking at the fetch though, I see this port is trying to fetch boost. Why is it doing this, instead of just using the macports version ?

File: databases/mysql8/Portfile [hunk 1]
```diff
@@ -0,0 +1,289 @@
...
+    distname            mysql-${version}
+
+    distfiles           ${distname}${extract.suffix}:mysql \
+                        ${boost_distname}${extract.suffix}:boost
+
+    checksums           ${distname}${extract.suffix} \
+                        rmd160  2e0d3326331b74fec8ab3e147f330d2d4f75382e \
+                        sha256  8d9fe89920dc8bbbde2857b7b877ad2fa5ec2f231c68e941d484f3b72735eaea \
+                        size    146811037 \
+                        ${boost_distname}${extract.suffix} \
```
Comment (cjones051073) [inline databases/mysql8/Portfile]: I don’t think we want this port to be fetching its own private boost version. Why can it not use the macports provided boost  instead ?

**Comment by herbygillot** [comment]
MySQL 8 is very sensitive to the Boost version.  As a matter of fact, its build process has facilities for downloading its required version on its own (`-DDOWNLOAD_BOOST`).  As of this moment, latest version of Boost available in MacPorts (v `1.66.0`) won't work, and we cannot update the MacPorts Boost willy-nilly due to the huge number of dependent ports.

This localized copy of Boost that gets downloaded by the MySQL 8 build process is used entirely and only for building MySQL, and does not make it to the MacPorts filesystem or affect anything else.  In this way, each successive version of MySQL can build with exactly the version it needs without needing to wait for MacPorts' Boost, or affect any of the other Boost-dependent ports.

**Comment by cjones051073** [comment]
> MySQL 8 is very sensitive to the Boost version. As a matter of fact, its build process has facilities for downloading its required version on its own (`-DDOWNLOAD_BOOST`). As of this moment, latest version of Boost available in MacPorts (v `1.66.0`) won't work, and we cannot update the MacPorts Boost willy-nilly due to the huge number of dependent ports.
> 
> This localized copy of Boost that gets downloaded by the MySQL 8 build process is used entirely and only for building MySQL, and does not make it to the MacPorts filesystem or affect anything else. In this way, each successive version of MySQL can build with exactly the version it needs without needing to wait for MacPorts' Boost, or affect any of the other Boost-dependent ports.

Ok, I see. As you probably are aware the boost update in macports is  being discussed in https://github.com/macports/macports-ports/pull/4243. The local use of boost here is I guess OK for now, but once the boost update  is merged, I think this port should still look to use it instead of download its own version.

**Comment by herbygillot** [comment]
> The local use of boost here is I guess OK for now, but once the boost update is merged, I think this port should still look to use it instead of download its own version.

Agreed 100%, that makes total sense.  I'll mark change request on the boost fetch as resolved.

File: databases/mysql8/Portfile (hunk 1 above)
Comment (herbygillot) [reply databases/mysql8/Portfile]: (resolved as per discussion)

**Comment by cjones051073** [comment]
> > The local use of boost here is I guess OK for now, but once the boost update is merged, I think this port should still look to use it instead of download its own version.
> 
> Agreed 100%, that makes total sense. I'll mark the change request on the boost fetch as resolved.

It would also be better to use -DDOWNLOAD_BOOST:BOOL=ON and let the build system handle the download, rather than the portfile explicitly doing it itself.

**Comment by herbygillot** [comment]
> It would also be better to use -DDOWNLOAD_BOOST:BOOL=ON and let the build system handle the download, rather than the portfile explicitly doing it itself.

I think it's better to have every distfile we need to build the software listed explicitly in the Portfile, as we can ensure integrity via checksums.  Additionally if something goes wrong with the download, the port build process fails in the appropriate phase (`fetch`), as opposed to the `build` phase.

File: databases/mysql8/Portfile [hunk 2]
```diff
@@ -0,0 +1,287 @@
...
+PortSystem              1.0
+
+name                    mysql8
+version                 8.0.16
+set boost_version       1.69.0
+categories              databases
+platforms               darwin
+license                 GPL-2
+maintainers             {gmail.com:herby.gillot @herbygillot} openmaintainer
+homepage                http://www.mysql.com/
```
Comment (mf2k) [inline databases/mysql8/Portfile]: Please use https here.

Comment (herbygillot) [reply databases/mysql8/Portfile]: Fixed.

**Comment by mf2k** [comment]
Thank you!

**Comment by herbygillot** [comment]
Thank you @mf2k, @cjones051073

**Pull request merged**